External/AvalonTools/CMakeLists.txt

```diff
@@ -3,103 +3,79 @@ if(NOT RDK_BUILD_AVALON_SUPPORT)
-  # apply patch to AvalonTools
-  configure_file("${fileToPatch}" "${fileToPatch}.orig" COPYONLY)
-  file(READ "${fileToPatch}" buffer)
-  string(REGEX REPLACE "MyFree\\(\\(char \\*\\)tempdir\\);"
-    "//MyFree((char *)tempdir);" buffer "${buffer}")
-  file(WRITE "${fileToPatch}" "${buffer}")
 endif()
 
 if (MSVC)
   add_definitions("/D_CRT_SECURE_NO_WARNINGS")
   add_compile_options(/wd4224 /wd4101 /wd4018 /wd4996 /wd4244 /wd4305 /wd4013 /wd4146 /wd4334 /wd4715 /wd4715 /nologo)
 endif(MSVC)
 
-set(avalon_clib_srcs ${AVALON_SRC_PATH}/layout.c
-	${AVALON_SRC_PATH}/symboltable.c
-	${AVALON_SRC_PATH}/patclean.c
-	${AVALON_SRC_PATH}/utilities.c
-	${AVALON_SRC_PATH}/symbol_lists.c
-	${AVALON_SRC_PATH}/stereo.c
-	${AVALON_SRC_PATH}/set.c
-	${AVALON_SRC_PATH}/perceive.c
-	${AVALON_SRC_PATH}/local.c
-	${AVALON_SRC_PATH}/graph.c
-	${AVALON_SRC_PATH}/geometry.c
-	${AVALON_SRC_PATH}/forio.c
-	${AVALON_SRC_PATH}/depictutil.c
-	${AVALON_SRC_PATH}/denormal.c
-	${AVALON_SRC_PATH}/casutils.c
-	${AVALON_SRC_PATH}/ssmatch.c
-	${AVALON_SRC_PATH}/rtutils.c
-	${AVALON_SRC_PATH}/smi2mol.c
-	${AVALON_SRC_PATH}/didepict.c
-	${AVALON_SRC_PATH}/pattern.c
-	${AVALON_SRC_PATH}/canonizer.c
-	${AVALON_SRC_PATH}/aacheck.c
-	${AVALON_SRC_PATH}/fixcharges.c
-	${AVALON_SRC_PATH}/struchk.c
-	${AVALON_SRC_PATH}/reaccsio.c
-	${AVALON_SRC_PATH}/hashcode.c
+set(avalon_clib_srcs ${AVALON_COMMON_PATH}/layout.c
+	${AVALON_COMMON_PATH}/symboltable.c
+	${AVALON_COMMON_PATH}/patclean.c
+	${AVALON_COMMON_PATH}/utilities.c
+	${AVALON_COMMON_PATH}/symbol_lists.c
+	${AVALON_COMMON_PATH}/stereo.c
+	${AVALON_COMMON_PATH}/set.c
+	${AVALON_COMMON_PATH}/perceive.c
+	${AVALON_COMMON_PATH}/local.c
+	${AVALON_COMMON_PATH}/graph.c
+	${AVALON_COMMON_PATH}/geometry.c
+	${AVALON_COMMON_PATH}/forio.c
+	${AVALON_COMMON_PATH}/depictutil.c
+	${AVALON_COMMON_PATH}/denormal.c
+	${AVALON_COMMON_PATH}/casutils.c
+	${AVALON_COMMON_PATH}/ssmatch.c
+	${AVALON_COMMON_PATH}/rtutils.c
+	${AVALON_COMMON_PATH}/smi2mol.c
+	${AVALON_COMMON_PATH}/didepict.c
+	${AVALON_COMMON_PATH}/pattern.c
+	${AVALON_COMMON_PATH}/canonizer.c
+	${AVALON_COMMON_PATH}/aacheck.c
+	${AVALON_COMMON_PATH}/fixcharges.c
+	${AVALON_PROGRAMS_PATH}/struchk.c
+	${AVALON_COMMON_PATH}/reaccsio.c
+	${AVALON_COMMON_PATH}/hashcode.c
   )
 
-# we need this to ensure that builds continue
-# to work on linux systems with older versions
-# of glibc when we're building with gcc-4.1.
-# Without this flag, we'll endup requiring
-# glibc 2.7.
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    add_definitions(-D_GNU_SOURCE=1)
-endif()
-
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-return-type -Wno-implicit-function-declaration -Wno-absolute-value -Wno-parentheses -Wno-logical-op-parentheses -Wno-dangling-else -Wno-format")
-endif()
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-result -Wformat-overflow=0 -Wformat=0 -Wno-format-security -Wno-implicit-function-declaration")
-endif()
```

Comment:
I'd prefer to have these warnings suppressed unless there's a good reason to remove them.

The builds now have a bunch of extra noise in them which add no value since the warnings are not in code we control.

Reply:
@greglandrum In general I prefer to have no warnings to suppressed warnings :-) I have already sent a patch to Bernd to suppress those warnings in the Avalon code.
That said, I am happy to reinstate the warning suppression until those are fixed - I'll update my PR shortly.

Reply:
> @greglandrum In general I prefer no warnings to suppressed warnings :-) I have already sent a patch to Bernd to suppress those warnings in the Avalon code. That said, I am happy to reinstate the warning suppression until those are fixed - I'll update my PR shortly.

I agree completely, but until the patch is actually merged, I'd rather be suppressing things. Thanks for agreeing to update.